urlcf7851e92b_Triploblastic_CSCD350_FinalProject_tgz-pJ8-MazeGame_MazeJ8/src/MazeGeneration/EnumMaze.java

```diff
@@ -0,0 +1,172 @@
+package MazeGeneration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Random;
+
+import MazeRoomLogic.MazeEnums.RoomType;
+
+/* Richard Sipes 
+ * CSCD350
+ * Generates a 2D maze with RoomType enums: START, EXIT, DOOR, PATH. 
+ * Includes start/exit coordinates and a ArrayStack of trace coordinates.
+ */
+
+public class EnumMaze {
+	private int size;
+	private int[] start, exit;
+	private RoomType[][] maze;
+	private ArrayStack<int[]> trace = new ArrayStack();
+	
+	public EnumMaze(int num) {
+		if (num % 2 == 0) 
+			throw new IllegalArgumentException("Maze size must be an odd number.");
+		size = num;
+		do {
+			randomPath(maze = initMaze(size), start = randomEdgeLoc(size), trace);
+			maze[start[0]][start[1]] = RoomType.START;
+			exit = trace.peek();
+			maze[exit[0]][exit[1]] = RoomType.EXIT;
+		} while (start[0] == exit[0] && start[1] == exit[1]); //generate mazes until start != exit (usually on first try)
+	}
+	
+	/*getters*/
+	public RoomType[][] getMaze() {return this.maze;}
+	public int getSize() {return this.size;}
+	public int[] getStart() {return this.start;}
+	public int[] getExit() {return this.exit;}
+	public ArrayStack<int[]> getTrace() {return this.trace;}
+	
+	
+	private static void randomPath(RoomType[][] maze, int[] start, ArrayStack<int[]> trace) {
+		recursion(start[0],start[1], maze, trace);	
+	}
+	
+	/*First-depth-search recursive method modified from  http://www.migapro.com/depth-first-search/ */
+	/*advances the path 2 squares & pushes coordinates to the trace stack*/
+	/*doors randomly placed before backtracking*/
+	private static void recursion(int r, int c, RoomType[][] maze, ArrayStack<int[]> trace) { 
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.